File: app/src/main/java/com/paradigmadigital/dagger/di/ApplicationComponent.java
package com.paradigmadigital.dagger.di;

import com.paradigmadigital.dagger.platform.ApplicationModule;
import com.paradigmadigital.dagger.ui.IAppCollaborator;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Component;
import eu.sesma.core.CoreCollaborator;

@Singleton
@Component(modules = {
        ApplicationModule.class
})
public interface ApplicationComponent {

    //Exposed to sub-graphs
    Context provideContext();

    IAppCollaborator provideAppCollaborator();
}

File: app/src/main/java/eu/sesma/dagger/ApplicationComponent.kt
package eu.sesma.dagger

import android.app.Application
import dagger.Component
import eu.sesma.dagger.core.CoreComponent

@ApplicationScoped
@Component(
        modules = [ApplicationModule::class],
        dependencies = [CoreComponent::class])
interface ApplicationComponent {

    //Exposed to sub-graphs
    fun provideContext(): Application

}
